### PR TITLE
[PBI & Workspace & Apps] - Feat/automatically load powerbi pages

### PR DESCRIPTION
**Pull request by jaysencpp**
# Description

### Automatically load pages
We can use a hook from the PBI package to retrieve all pages of the report __after__ the PBI component is loaded. We need to do the API calls to retrieve that information first.

To let the parent components of the PBI component know about the report, I added a callback function that takes the embedded object (report object) as an argument and let the parent components do the rest.

There is no need to declare all the page ids and titles, so all of those are now removed.

### Bookmarks
A couple of things about bookmarks update: There is a bug if you use the "Portal Bookmarks sidesheet" while inside a Workspace/PowerBIViewer where it doesn't apply the bookmark state to the report. It's because the sidesheet doesn't use the events so PowerBI doesn't fire the `applyBookmark` function. A workaround is to apply the state automatically. We can do this since the outside components have access to the report object, and we know if the bookmarks comes externally (sidesheet) or internally (bookmark dropdown inside the workspace).

### Misc
Some background styling for PBI.

Completes: [AB#231675](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/231675)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/PowerBiViewer/src/PowerBiViewer.tsx [hunk 1]
```diff
@@ -1,132 +1,56 @@
...
-                    />
-                )}
-            </ContentWrapper>
+            <PowerBiViewerHeader {...props} groupName={props.groupe} />
+
+            {configReport && configReport.reportURI && ready && (
+                <PowerBI
+                    reportUri={configReport.reportURI}
+                    filterOptions={configReport.filter}
+                    onReportLoad={async (report) => {
```
Comment (jaysencpp) [inline src/Core/PowerBiViewer/src/PowerBiViewer.tsx:40]: skal ikke være async

File: src/Core/WorkSpace/src/Context/ViewProvider.tsx [hunk 2]
```diff
@@ -43,19 +45,6 @@ export const ViewProvider = ({ children }: PropsWithChildren<unknown>): JSX.Elem
...
-                hasPowerBi: true,
-                pages: powerBiOptions.pages,
-                activePage: powerBiOptions.pages.find((p) => p.default),
-            }));
-        } else {
-            setState(INIT_STATE);
-        }
-    }, [powerBiOptions]);
-
     const registerPages = useCallback((pages: Page[]) => {
```
Comment (jaysencpp) [inline src/Core/WorkSpace/src/Context/ViewProvider.tsx:48]: fjern?

File: src/Core/WorkSpace/src/Context/ViewProvider.tsx [hunk 3]
```diff
@@ -78,6 +67,36 @@ export const ViewProvider = ({ children }: PropsWithChildren<unknown>): JSX.Elem
...
+                console.error('Cannot set report');
+            }
+        });
+    };
+
+    useEffect(() => {
+        if (powerBiOptions !== undefined) {
+            setState((s) => ({
+                ...s,
+                hasPowerBi: true,
```
Comment (jaysencpp) [inline src/Core/WorkSpace/src/Context/ViewProvider.tsx:94]: se hva den blir brukt til

File: src/Core/WorkSpace/src/Tabs/PowerBiTab.tsx [hunk 4]
```diff
@@ -15,14 +15,17 @@ export const Wrapper = styled.div`
...
-    const { activePage, pbiOptions } = useViewerContext();
+    const { activePage, pbiOptions, setPbiReport } = useViewerContext();
     const { applyBookmark, saveBookmark } = useBookmarkContext<PowerBIBookmarkPayload>();
     if (powerBiOptions) {
         return (
             <Wrapper>
                 <PowerBI
                     reportUri={powerBiOptions?.reportURI}
                     filterOptions={powerBiOptions.filter}
+                    onReportLoad={async (report) => {
```
Comment (jaysencpp) [inline src/Core/WorkSpace/src/Tabs/PowerBiTab.tsx:26]: skal ikke være async

**Review by Andreas-Tonnessen — APPROVED** [review]

**Pull request merged**